runtime/sema/check_assignment.go

```diff
@@ -294,17 +296,26 @@ func (checker *Checker) visitMemberExpressionAssignment(
 
 			if !functionActivation.ReturnInfo.MaybeReturned {
 
-				// If the field is constant and it has already previously been
-				// initialized, report an error for the repeated assignment
+				// If the field is constant,
+				// or it is variable and resource-kinded,
+				// and it has already previously been initialized,
+				// report an error for the repeated assignment / initialization
+				//
+				// Assigning to a variable, resource-kinded field is invalid,
+				// because the initial value would get lost.
```

Comment:
It might be clearer to use more well-named variables than to have a large comment. You started this with `targetIsConstant`.

Reply:
What is your concrete suggestion?

Reply:
On second thought, it wouldn't be good to run `initializedFieldMembers.Contains(accessedSelfMember)` if unnecessary. It could be put into a function but that's probably less clear overall.

Anyway, here's what I was thinking:

```
...

targetIsConstant := member.VariableKind == ast.VariableKindConstant
targetIsResource := member.TypeAnnotation.Type.IsResourceType()
targetIsInitialized := functionActivation.InitializationInfo.InitializedFieldMembers.Contains(accessedSelfMember)

...

// Report error for repeated assignment/initialization of constants or resource variables.
// Cannot assign to resource-kinded fields because the initial value would be lost.

if (targetIsConstant || targetIsResource) && targetIsInitialized {

...
```